coders/tiff.c

```diff
@@ -218,6 +218,34 @@ static MagickBooleanType
   WriteGROUP4Image(const ImageInfo *,Image *),
   WritePTIFImage(const ImageInfo *,Image *),
   WriteTIFFImage(const ImageInfo *,Image *);
+
+static void InitPSDInfo(Image *image, Image *layer, PSDInfo *info)
+{
+  info->version=1;
+  info->columns=layer->columns;
+  info->rows=layer->rows;
+  /* Setting the mode to a value that won't change the colorspace */
+  info->mode=10;
+  /* Assume that image has matte */
+  if (IsGrayImage(image,&image->exception) != MagickFalse)
+    info->channels=2U;
+  else
+    if (image->storage_class == PseudoClass)
+      {
+        info->mode=2;
+        info->channels=2U;
```

Comment:
This should always be set to `1`. There won't be an alpha channel when `image->storage_class == PseudoClass`. I think it would be better to swap this with the `if (IsGrayImage(image,&image->exception) != MagickFalse)`.

Reply:
And it looks like you need to rebase your branch so we can get this merged.